components/core/src/service.rs

```diff
@@ -13,10 +13,10 @@ use std::{fmt,
 
 lazy_static::lazy_static! {
     static ref SG_FROM_STR_RE: Regex =
-        Regex::new(r"\A((?P<application_environment>[^#@]+)#)?(?P<service>[^#@.]+)\.(?P<group>[^#@.]+)(@(?P<organization>[^#@.]+))?\z").unwrap();
+        Regex::new(r"\A(([^#@]+)#)?([^#@]+)\.([^#@.]+)(@([^#@.]+))?\z").unwrap();
```

Comment:
Actually, as long as we're keeping a regex-based implementation, the named capture groups make this complicated regex _easier_ to understand, and also make subsequent processing of it easier to read (e.g. `captures.name("service")` is better than `captures.get(2)`) 

I think a better way to clean this up would be to take advantage of insignificant whitespace mode (see in https://docs.rs/regex/1.1.6/regex/#example-replacement-with-named-capture-groups), which would allow us to split each component of this regex onto its own line for readability and comprehensibility.